sql/hive/src/main/scala/org/apache/spark/sql/hive/execution/HiveSqlParser.scala

```diff
@@ -113,7 +113,10 @@ class HiveSqlAstBuilder extends SparkSqlAstBuilder {
     if (ctx.REPLICATION != null) {
       logWarning("REPLICATION clause is ignored.")
     }
-    DropTable(visitTableIdentifier(ctx.tableIdentifier).toString, ctx.EXISTS != null)
+    DropTable(
+      visitTableIdentifier(ctx.tableIdentifier),
+      ctx.EXISTS != null,
```

Comment:
We should support the syntax to remain compatible with Hive. We should add a warning here if we don't add `PURGE` functionality.